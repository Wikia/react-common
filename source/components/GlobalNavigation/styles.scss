// import variables
@import "~design-system/dist/scss/wds-functions/index.scss";
@import "~design-system/dist/scss/wds-mixins/index.scss";
@import "~design-system/dist/scss/wds-variables/index.scss";

// import wds-button
@import "~design-system/dist/scss/wds-components/_global-navigation.scss";
@import "~design-system/dist/scss/wds-components/_sign-out.scss";
@import "~design-system/dist/scss/wds-components/_search-modal.scss";
@import "~design-system/dist/scss/wds-components/_is-hidden.scss";
@import "~design-system/dist/scss/wds-components/global-navigation/_user-modal.scss";

.wds-global-navigation {
	//position: static;
}
