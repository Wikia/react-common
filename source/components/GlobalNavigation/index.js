import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import Icon from '../Icon/index';
import Button from '../Button/index';

import LogoFandomWhite from '../../assets/LogoFandomWhite';

import GlobalNavigationSearch from './components/GlobalNavigationSearch/GlobalNavigationSearch';
import GlobalNavigationLinkText from './components/GlobalNavigationLink/GlobalNavigationLinkText';
import GlobalNavigationLinkGroup from './components/GlobalNavigationLink/GlobalNavigationLinkGroup';
import GlobalNavigationLinkButton from './components/GlobalNavigationLink/GlobalNavigationLinkButton';
import GlobalNavigationUser from './components/GlobalNavigationUser/GlobalNavigationUser';
import GlobalNavigationSearchModal from './components/GlobalNavigationSearch/GlobalNavigationSearchModal';
import GlobalNavigationMobileUser from './components/GlobalNavigationUser/GlobalNavigationMobileUser';
import NotificationsDataProvider from './components/GlobalNavigationNotifications/NotificationsDataProvider';
import NotificationsDropdown from './components/GlobalNavigationNotifications/NotificationsDropdown';

import { NotificationsConsumer } from './utils/NotificationContext';

import './styles.scss';

/* eslint-disable react/no-array-index-key */
class GlobalNavigation extends React.Component {
    constructor(props) {
        super(props);

        this.openModal = this.openModal.bind(this);
        this.closeModal = this.closeModal.bind(this);
    }

    state = {
        isSearchModalOpen: false,
        isUserModalOpen: false,
    };

    openModal(type) {
        if (type === 'search') {
            this.setState({ isSearchModalOpen: true });
        }

        if (type === 'user') {
            this.setState({ isUserModalOpen: true });
        }
    }

    closeModal() {
        this.setState({
            isSearchModalOpen: false,
            isUserModalOpen: false,
        });
    }

    renderMainNavigation(navigation) {
        return navigation.map((link, index) => {
            if (link.type === 'link-text') {
                return <GlobalNavigationLinkText key={index} link={link} isStandaloneLink />;
            }

            if (link.type === 'link-group') {
                return <GlobalNavigationLinkGroup key={index} link={link} />;
            }

            return null;
        });
    }

    render() {
        const {
            model,
            onSearchToggleClicked,
            onSearchSuggestionChosen,
            goToSearchResults,
            track,
            onSearchCloseClicked,
        } = this.props;
        const { isSearchModalOpen, isUserModalOpen } = this.state;
        const containerClass = classNames('wds-global-navigation', {
            'wds-search-is-active': isSearchModalOpen,
            'wds-is-modal-opened': isSearchModalOpen || isUserModalOpen,
        });

        return (
            <NotificationsDataProvider serviceUrl={model['services-domain']}>
                <div className={containerClass}>
                    <div className="wds-global-navigation__content-bar-left">
                        <a
                            href={model.logo.href}
                            className="wds-global-navigation__logo"
                            data-tracking-label={model.logo['tracking-label']}
                        >
                            <LogoFandomWhite height="27" />
                        </a>
                        <nav className="wds-global-navigation__links">
                            {this.renderMainNavigation(model['main-navigation'])}
                        </nav>
                    </div>
                    <div className="wds-global-navigation__content-bar-right">
                        <div className="wds-global-navigation__dropdown-controls">
                            <GlobalNavigationSearch
                                model={model.search}
                                onSearchToggleClicked={onSearchToggleClicked}
                                onSearchSuggestionChosen={onSearchSuggestionChosen}
                                onSearchCloseClicked={onSearchCloseClicked}
                                goToSearchResults={goToSearchResults}
                                track={track}
                            />
                            <GlobalNavigationUser data={model} />
                            {
                                model.user && (
                                    <NotificationsConsumer>
                                        {
                                            ({ unreadCount, loadFirstPage, markAllAsRead }) => (
                                                <NotificationsDropdown
                                                    track={track}
                                                    loadFirstPage={loadFirstPage}
                                                    unreadCount={unreadCount}
                                                    markAllAsRead={markAllAsRead}
                                                />
                                            )
                                        }
                                    </NotificationsConsumer>
                                )
                            }
                            <div className="wds-global-navigation__start-a-wiki">
                                <GlobalNavigationLinkButton link={model['create-wiki']} />
                            </div>
                        </div>
                        <div className="wds-global-navigation__modal-controls">
                            <GlobalNavigationSearchModal
                                model={model}
                                isOpen={isSearchModalOpen}
                                openModal={this.openModal}
                            >
                                <GlobalNavigationSearch
                                    model={model.search}
                                    onSearchToggleClicked={onSearchToggleClicked}
                                    onSearchCloseClicked={onSearchCloseClicked}
                                    onSearchSuggestionChosen={onSearchSuggestionChosen}
                                    goToSearchResults={goToSearchResults}
                                    track={track}
                                    inSearchModal
                                />
                                <nav className="wds-global-navigation__links">
                                    {this.renderMainNavigation(model['main-navigation'])}
                                </nav>
                            </GlobalNavigationSearchModal>
                            <GlobalNavigationMobileUser
                                data={model}
                                openModal={this.openModal}
                                isOpen={isUserModalOpen}
                                track={track}
                            />
                            <Button
                                onClick={this.closeModal}
                                className="wds-global-navigation__modal-control wds-global-navigation__modal-control-close"
                                text
                            >
                                <Icon name="close" />
                            </Button>
                        </div>
                    </div>
                </div>
            </NotificationsDataProvider>
        );
    }
}

GlobalNavigation.propTypes = {
    goToSearchResults: PropTypes.func,
    // eslint-disable-next-line react/forbid-prop-types
    model: PropTypes.object.isRequired,
    onSearchCloseClicked: PropTypes.func,
    onSearchSuggestionChosen: PropTypes.func,
    onSearchToggleClicked: PropTypes.func,
    track: PropTypes.func,
};

GlobalNavigation.defaultProps = {
    goToSearchResults: () => {},
    onSearchCloseClicked: () => {},
    onSearchSuggestionChosen: () => {},
    onSearchToggleClicked: () => {},
    track: () => {},
};

export default GlobalNavigation;
