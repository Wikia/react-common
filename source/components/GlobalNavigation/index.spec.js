import React from 'react';
import { shallow } from 'enzyme';
import merge from 'lodash/merge';

import GlobalNavigation from './index';

const defaultProps = {
    model: {
        logo: {
            type: 'link-image',
            href: '//fandom.wikia.com/',
            'image-data': { type: 'wds-svg', name: 'wds-company-logo-fandom-white' },
            'tracking-label': 'logo',
        },
        search: { search: 'mock' },
        'create-wiki': { 'create-wiki': 'mock' },
        'main-navigation': [{
            type: 'link-text',
            title: { type: 'translatable-text', key: 'global-navigation-fandom-overview-link-vertical-games' },
            href: '//fandom.wikia.com/topics/games',
            'tracking-label': 'link.games',
        }, {
            type: 'link-other',
            title: { type: 'translatable-text', key: 'global-navigation-fandom-overview-link-vertical-games' },
            href: '//fandom.wikia.com/topics/games',
            'tracking-label': 'link.blah',
        }, {
            type: 'link-group',
            title: { type: 'translatable-text', key: 'global-navigation-wikis-header' },
            'tracking-label': 'link.wikis',
            items: [{
                type: 'link-text',
                title: { type: 'translatable-text', key: 'global-navigation-wikis-explore' },
                href: '//fandom.wikia.com/explore',
                'tracking-label': 'link.explore',
            }],
        }],
        user: { user: 'mock' },
        notifications: { notifications: 'mock' },
        'services-domain': 'https://services.wikia.com/',
        'partner-slot': { partner: 'mock' },
    },
    siteName: { siteName: 'mock' },
    i18nNamespace: 'designSystem',
};

function renderComponent(props, options = {}) {
    const computedProps = merge({}, defaultProps, props);

    return shallow(<GlobalNavigation {...computedProps} />, { disableLifecycleMethods: true, ...options });
}

describe('render', () => {
    test('GlobalNavigation renders correctly', () => {
        expect(renderComponent()).toMatchSnapshot();
    });

    test('GlobalNavigation renders correctly when user or partnerSlot are not defined', () => {
        expect(renderComponent({ model: { user: null, 'partner-slot': null } })).toMatchSnapshot();
    });
});

describe('componentDidMount / componentWillUnmount', () => {
    test('sets communityBar property properly, handles event listeners', () => {
        const elMock = document.createElement('div');
        const addEventListenerMock = jest.fn();
        const removeEventListenerMock = jest.fn();
        const onScrollMock = jest.fn();
        const wrapper = renderComponent();
        const instance = wrapper.instance();

        global.document.querySelector = () => elMock;
        global.window.addEventListener = addEventListenerMock;
        global.window.removeEventListener = removeEventListenerMock;
        instance.onScroll = onScrollMock;

        instance.componentDidMount();

        expect(instance.communityBar).toBe(elMock);
        expect(addEventListenerMock).toBeCalledWith('scroll', onScrollMock);

        instance.componentWillUnmount();

        expect(removeEventListenerMock).toBeCalledWith('scroll', onScrollMock);
    });
});

describe('onRedirectToSearchResults', () => {
    test('calls methods', () => {
        const closeAndDeactivateMock = jest.fn();
        const goToSearchResultsMock = jest.fn();
        const queryMock = 'queryMock';
        const wrapper = renderComponent({ goToSearchResults: goToSearchResultsMock });
        const instance = wrapper.instance();

        instance.closeAndDeactivate = closeAndDeactivateMock;

        instance.onRedirectToSearchResults(queryMock);

        expect(closeAndDeactivateMock).toBeCalledWith();
        expect(goToSearchResultsMock).toBeCalledWith(queryMock);
    });

    test('calls default prop when goToSearchResults is not passed', () => {
        const closeAndDeactivateMock = jest.fn();
        const queryMock = 'queryMock';
        const wrapper = renderComponent();
        const instance = wrapper.instance();

        instance.closeAndDeactivate = closeAndDeactivateMock;

        instance.onRedirectToSearchResults(queryMock);

        expect(closeAndDeactivateMock).toBeCalledWith();
    });
});

describe('onSearchActivation', () => {
    test('calls methods, updates state', () => {
        const onSearchToggleClickedMock = jest.fn();
        const wrapper = renderComponent({ onSearchToggleClicked: onSearchToggleClickedMock });
        const instance = wrapper.instance();

        instance.onSearchActivation();

        expect(wrapper.state('isSearchExpanded')).toBe(true);
        expect(onSearchToggleClickedMock).toBeCalledWith();
    });

    test('calls default prop when onSearchToggleClicked is not passed, updates state', () => {
        const wrapper = renderComponent();
        const instance = wrapper.instance();

        instance.onSearchActivation();

        expect(wrapper.state('isSearchExpanded')).toBe(true);
    });
});

describe('onSearchClose', () => {
    test('calls methods', () => {
        const closeAndDeactivateMock = jest.fn();
        const onSearchCloseClickedMock = jest.fn();
        const wrapper = renderComponent({ onSearchCloseClicked: onSearchCloseClickedMock });
        const instance = wrapper.instance();

        instance.closeAndDeactivate = closeAndDeactivateMock;

        instance.onSearchClose();

        expect(closeAndDeactivateMock).toBeCalledWith();
        expect(onSearchCloseClickedMock).toBeCalledWith();
    });

    test('calls default prop when onSearchCloseClicked is not passed', () => {
        const closeAndDeactivateMock = jest.fn();
        const wrapper = renderComponent();
        const instance = wrapper.instance();

        instance.closeAndDeactivate = closeAndDeactivateMock;

        instance.onSearchClose();

        expect(closeAndDeactivateMock).toBeCalledWith();
    });
});

describe('onSearchSuggestionChosen', () => {
    test('calls methods', () => {
        const argsMock = ['suggestion', [], 123];
        const closeAndDeactivateMock = jest.fn();
        const onSearchSuggestionChosenMock = jest.fn();
        const wrapper = renderComponent({ onSearchSuggestionChosen: onSearchSuggestionChosenMock });
        const instance = wrapper.instance();

        instance.closeAndDeactivate = closeAndDeactivateMock;

        instance.onSearchSuggestionChosen(...argsMock);

        expect(closeAndDeactivateMock).toBeCalledWith();
        expect(onSearchSuggestionChosenMock).toBeCalledWith(...argsMock);
    });

    test('calls default prop when onSearchSuggestionChosen is not passed', () => {
        const argsMock = ['suggestion', [], 123];
        const closeAndDeactivateMock = jest.fn();
        const wrapper = renderComponent();
        const instance = wrapper.instance();

        instance.closeAndDeactivate = closeAndDeactivateMock;

        instance.onSearchSuggestionChosen(...argsMock);

        expect(closeAndDeactivateMock).toBeCalledWith();
    });
});

describe('onSearchSuggestionsImpression', () => {
    test('calls methods', () => {
        const argsMock = [[], 123];
        const onSearchSuggestionsImpressionMock = jest.fn();
        const wrapper = renderComponent({ onSearchSuggestionsImpression: onSearchSuggestionsImpressionMock });
        const instance = wrapper.instance();

        instance.onSearchSuggestionsImpression(...argsMock);

        expect(onSearchSuggestionsImpressionMock).toBeCalledWith(...argsMock);
    });

    test('calls default prop when onSearchSuggestionsImpression is not passed', () => {
        const argsMock = [[], 123];
        const wrapper = renderComponent();
        const instance = wrapper.instance();

        instance.onSearchSuggestionsImpression(...argsMock);

        // no expect
    });
});

describe('onTrackingLabelClick', () => {
    test('calls track prop when clicked element has some parent with proper attribute', () => {
        const trackMock = jest.fn();
        const eventMock = { target: { closest: () => ({ getAttribute: () => 'blah' }) } };
        const wrapper = renderComponent({ track: trackMock });
        const instance = wrapper.instance();

        instance.onTrackingLabelClick(eventMock);

        expect(trackMock).toBeCalledWith(expect.any(Object));
    });

    test('calls default prop when track is not passed and clicked element has some parent with proper attribute', () => {
        const eventMock = { target: { closest: () => ({ getAttribute: () => 'blah' }) } };
        const wrapper = renderComponent();
        const instance = wrapper.instance();

        instance.onTrackingLabelClick(eventMock);

        // no expect
    });

    test('does not call track prop when clicked element does not have any parent with proper attribute', () => {
        const trackMock = jest.fn();
        const eventMock = { target: { closest: () => null } };
        const wrapper = renderComponent({ track: trackMock });
        const instance = wrapper.instance();

        instance.onTrackingLabelClick(eventMock);

        expect(trackMock).not.toBeCalled();
    });
});

describe('onScroll', () => {
    test('does not update state when communityBar element does not exist', () => {
        const wrapper = renderComponent();
        const instance = wrapper.instance();

        instance.communityBar = null;
        instance.setState = jest.fn();

        instance.onScroll();

        expect(instance.setState).not.toBeCalled();
    });

    test('does not update state when isCommunityBarActive is false and view is scrolled above that element', () => {
        const wrapper = renderComponent();
        const instance = wrapper.instance();

        instance.communityBar = { getBoundingClientRect: () => ({ top: 10 }) };
        instance.nav = { current: { getBoundingClientRect: () => ({ top: 0 }) } };
        instance.setState = jest.fn();

        instance.onScroll();

        expect(instance.setState).not.toBeCalled();
    });

    test('updates state when communityBar element exist and view is scrolled below that element', () => {
        const wrapper = renderComponent();
        const instance = wrapper.instance();

        instance.communityBar = { getBoundingClientRect: () => ({ top: -10 }) };
        instance.nav = { current: { getBoundingClientRect: () => ({ top: 0 }) } };

        instance.onScroll();

        expect(wrapper.state('isCommunityBarActive')).toBe(true);
    });

    test('updates state when communityBar element exist and view is scrolled above that element', () => {
        const wrapper = renderComponent().setState({ isCommunityBarActive: true });
        const instance = wrapper.instance();

        instance.communityBar = { getBoundingClientRect: () => ({ top: 10 }) };
        instance.nav = { current: { getBoundingClientRect: () => ({ top: 0 }) } };

        instance.onScroll();

        expect(wrapper.state('isCommunityBarActive')).toBe(false);
    });
});

describe('closeAndDeactivate', () => {
    test('updates state', () => {
        const stateMock = { isSearchModalOpen: true, isUserModalOpen: true, isSearchExpanded: true };
        const wrapper = renderComponent().setState(stateMock);
        const instance = wrapper.instance();

        instance.closeAndDeactivate();

        expect(wrapper.state('isSearchModalOpen')).toBe(false);
        expect(wrapper.state('isUserModalOpen')).toBe(false);
        expect(wrapper.state('isSearchExpanded')).toBe(false);
    });
});

describe('openModal', () => {
    test('calls onModalOpen prop and updates state when "search" type is passed', () => {
        const onModalOpenMock = jest.fn();
        const typeMock = 'search';
        const wrapper = renderComponent({ onModalOpen: onModalOpenMock });
        const instance = wrapper.instance();

        instance.openModal(typeMock);

        expect(wrapper.state('isSearchModalOpen')).toBe(true);
        expect(onModalOpenMock).toBeCalledWith(typeMock);
    });

    test('calls onModalOpen prop and updates state when "user" type is passed', () => {
        const onModalOpenMock = jest.fn();
        const typeMock = 'user';
        const wrapper = renderComponent({ onModalOpen: onModalOpenMock });
        const instance = wrapper.instance();

        instance.openModal(typeMock);

        expect(wrapper.state('isUserModalOpen')).toBe(true);
        expect(onModalOpenMock).toBeCalledWith(typeMock);
    });

    test('calls default prop when onModalOpen is not passed', () => {
        const typeMock = 'user';
        const wrapper = renderComponent();
        const instance = wrapper.instance();

        instance.openModal(typeMock);

        expect(wrapper.state('isUserModalOpen')).toBe(true);
    });
});
