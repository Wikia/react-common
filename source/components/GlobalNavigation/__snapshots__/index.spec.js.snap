// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render GlobalNavigation renders correctly 1`] = `
<ContextProvider
  value="designSystem"
>
  <NotificationsDataProvider
    isAuthenticated={true}
    serviceUrl="https://services.wikia.com/"
  >
    <div
      className="wds-global-navigation wds-has-partner-slot"
      onClick={[Function]}
    >
      <div
        className="wds-global-navigation__content-bar-left"
      >
        <a
          className="wds-global-navigation__logo"
          data-tracking-label="logo"
          href="//fandom.wikia.com/"
        >
          <_default
            className="wds-global-navigation__logo-image"
          />
        </a>
        <nav
          className="wds-global-navigation__links"
        >
          <LinkText
            isStandaloneLink={true}
            key="0"
            link={
              Object {
                "href": "//fandom.wikia.com/topics/games",
                "title": Object {
                  "key": "global-navigation-fandom-overview-link-vertical-games",
                  "type": "translatable-text",
                },
                "tracking-label": "link.games",
                "type": "link-text",
              }
            }
          />
          <LinkGroup
            key="2"
            link={
              Object {
                "items": Array [
                  Object {
                    "href": "//fandom.wikia.com/explore",
                    "title": Object {
                      "key": "global-navigation-wikis-explore",
                      "type": "translatable-text",
                    },
                    "tracking-label": "link.explore",
                    "type": "link-text",
                  },
                ],
                "title": Object {
                  "key": "global-navigation-wikis-header",
                  "type": "translatable-text",
                },
                "tracking-label": "link.wikis",
                "type": "link-group",
              }
            }
          />
        </nav>
      </div>
      <div
        className="wds-global-navigation__content-bar-right"
      >
        <CommunityBar
          model={
            Object {
              "href": "//fandom.wikia.com/",
              "image-data": Object {
                "name": "wds-company-logo-fandom-white",
                "type": "wds-svg",
              },
              "tracking-label": "logo",
              "type": "link-image",
            }
          }
          siteName={
            Object {
              "siteName": "mock",
            }
          }
        />
        <div
          className="wds-global-navigation__dropdown-controls"
        >
          <Search
            inSearchModal={false}
            isSearchExpanded={false}
            model={
              Object {
                "search": "mock",
              }
            }
            onRedirectToSearchResults={[Function]}
            onSearchActivation={[Function]}
            onSearchClose={[Function]}
            onSearchSuggestionChosen={[Function]}
            onSearchSuggestionsImpression={[Function]}
            track={[Function]}
          />
          <User
            data={
              Object {
                "create-wiki": Object {
                  "create-wiki": "mock",
                },
                "logo": Object {
                  "href": "//fandom.wikia.com/",
                  "image-data": Object {
                    "name": "wds-company-logo-fandom-white",
                    "type": "wds-svg",
                  },
                  "tracking-label": "logo",
                  "type": "link-image",
                },
                "main-navigation": Array [
                  Object {
                    "href": "//fandom.wikia.com/topics/games",
                    "title": Object {
                      "key": "global-navigation-fandom-overview-link-vertical-games",
                      "type": "translatable-text",
                    },
                    "tracking-label": "link.games",
                    "type": "link-text",
                  },
                  Object {
                    "href": "//fandom.wikia.com/topics/games",
                    "title": Object {
                      "key": "global-navigation-fandom-overview-link-vertical-games",
                      "type": "translatable-text",
                    },
                    "tracking-label": "link.blah",
                    "type": "link-other",
                  },
                  Object {
                    "items": Array [
                      Object {
                        "href": "//fandom.wikia.com/explore",
                        "title": Object {
                          "key": "global-navigation-wikis-explore",
                          "type": "translatable-text",
                        },
                        "tracking-label": "link.explore",
                        "type": "link-text",
                      },
                    ],
                    "title": Object {
                      "key": "global-navigation-wikis-header",
                      "type": "translatable-text",
                    },
                    "tracking-label": "link.wikis",
                    "type": "link-group",
                  },
                ],
                "notifications": Object {
                  "notifications": "mock",
                },
                "partner-slot": Object {
                  "partner": "mock",
                },
                "search": Object {
                  "search": "mock",
                },
                "services-domain": "https://services.wikia.com/",
                "user": Object {
                  "user": "mock",
                },
              }
            }
          />
          <NotificationsDropdown
            track={[Function]}
          />
          <div
            className="wds-global-navigation__start-a-wiki"
          >
            <LinkButton
              link={
                Object {
                  "create-wiki": "mock",
                }
              }
            />
          </div>
        </div>
        <div
          className="wds-global-navigation__modal-controls"
        >
          <SearchModal
            isOpen={false}
            openModal={[Function]}
          >
            <Search
              inSearchModal={true}
              isSearchExpanded={false}
              model={
                Object {
                  "search": "mock",
                }
              }
              onRedirectToSearchResults={[Function]}
              onSearchActivation={[Function]}
              onSearchClose={[Function]}
              onSearchSuggestionChosen={[Function]}
              onSearchSuggestionsImpression={[Function]}
              track={[Function]}
            />
            <nav
              className="wds-global-navigation__links"
            >
              <LinkText
                isStandaloneLink={true}
                key="0"
                link={
                  Object {
                    "href": "//fandom.wikia.com/topics/games",
                    "title": Object {
                      "key": "global-navigation-fandom-overview-link-vertical-games",
                      "type": "translatable-text",
                    },
                    "tracking-label": "link.games",
                    "type": "link-text",
                  }
                }
              />
              <LinkGroup
                key="2"
                link={
                  Object {
                    "items": Array [
                      Object {
                        "href": "//fandom.wikia.com/explore",
                        "title": Object {
                          "key": "global-navigation-wikis-explore",
                          "type": "translatable-text",
                        },
                        "tracking-label": "link.explore",
                        "type": "link-text",
                      },
                    ],
                    "title": Object {
                      "key": "global-navigation-wikis-header",
                      "type": "translatable-text",
                    },
                    "tracking-label": "link.wikis",
                    "type": "link-group",
                  }
                }
              />
            </nav>
          </SearchModal>
          <MobileUser
            data={
              Object {
                "create-wiki": Object {
                  "create-wiki": "mock",
                },
                "logo": Object {
                  "href": "//fandom.wikia.com/",
                  "image-data": Object {
                    "name": "wds-company-logo-fandom-white",
                    "type": "wds-svg",
                  },
                  "tracking-label": "logo",
                  "type": "link-image",
                },
                "main-navigation": Array [
                  Object {
                    "href": "//fandom.wikia.com/topics/games",
                    "title": Object {
                      "key": "global-navigation-fandom-overview-link-vertical-games",
                      "type": "translatable-text",
                    },
                    "tracking-label": "link.games",
                    "type": "link-text",
                  },
                  Object {
                    "href": "//fandom.wikia.com/topics/games",
                    "title": Object {
                      "key": "global-navigation-fandom-overview-link-vertical-games",
                      "type": "translatable-text",
                    },
                    "tracking-label": "link.blah",
                    "type": "link-other",
                  },
                  Object {
                    "items": Array [
                      Object {
                        "href": "//fandom.wikia.com/explore",
                        "title": Object {
                          "key": "global-navigation-wikis-explore",
                          "type": "translatable-text",
                        },
                        "tracking-label": "link.explore",
                        "type": "link-text",
                      },
                    ],
                    "title": Object {
                      "key": "global-navigation-wikis-header",
                      "type": "translatable-text",
                    },
                    "tracking-label": "link.wikis",
                    "type": "link-group",
                  },
                ],
                "notifications": Object {
                  "notifications": "mock",
                },
                "partner-slot": Object {
                  "partner": "mock",
                },
                "search": Object {
                  "search": "mock",
                },
                "services-domain": "https://services.wikia.com/",
                "user": Object {
                  "user": "mock",
                },
              }
            }
            isOpen={false}
            openModal={[Function]}
            track={[Function]}
          />
          <Button
            className="wds-global-navigation__modal-control wds-global-navigation__modal-control-close"
            disabled={false}
            fullwidth={false}
            href={null}
            light={false}
            onClick={[Function]}
            secondary={false}
            square={false}
            text={true}
          >
            <Icon
              className=""
              name="close"
              small={false}
              tiny={false}
            />
          </Button>
        </div>
      </div>
      <PartnerSlot
        model={
          Object {
            "partner": "mock",
          }
        }
      />
    </div>
  </NotificationsDataProvider>
</ContextProvider>
`;

exports[`render GlobalNavigation renders correctly when user or partnerSlot are not defined 1`] = `
<ContextProvider
  value="designSystem"
>
  <NotificationsDataProvider
    isAuthenticated={false}
    serviceUrl="https://services.wikia.com/"
  >
    <div
      className="wds-global-navigation"
      onClick={[Function]}
    >
      <div
        className="wds-global-navigation__content-bar-left"
      >
        <a
          className="wds-global-navigation__logo"
          data-tracking-label="logo"
          href="//fandom.wikia.com/"
        >
          <_default
            className="wds-global-navigation__logo-image"
          />
        </a>
        <nav
          className="wds-global-navigation__links"
        >
          <LinkText
            isStandaloneLink={true}
            key="0"
            link={
              Object {
                "href": "//fandom.wikia.com/topics/games",
                "title": Object {
                  "key": "global-navigation-fandom-overview-link-vertical-games",
                  "type": "translatable-text",
                },
                "tracking-label": "link.games",
                "type": "link-text",
              }
            }
          />
          <LinkGroup
            key="2"
            link={
              Object {
                "items": Array [
                  Object {
                    "href": "//fandom.wikia.com/explore",
                    "title": Object {
                      "key": "global-navigation-wikis-explore",
                      "type": "translatable-text",
                    },
                    "tracking-label": "link.explore",
                    "type": "link-text",
                  },
                ],
                "title": Object {
                  "key": "global-navigation-wikis-header",
                  "type": "translatable-text",
                },
                "tracking-label": "link.wikis",
                "type": "link-group",
              }
            }
          />
        </nav>
      </div>
      <div
        className="wds-global-navigation__content-bar-right"
      >
        <CommunityBar
          model={
            Object {
              "href": "//fandom.wikia.com/",
              "image-data": Object {
                "name": "wds-company-logo-fandom-white",
                "type": "wds-svg",
              },
              "tracking-label": "logo",
              "type": "link-image",
            }
          }
          siteName={
            Object {
              "siteName": "mock",
            }
          }
        />
        <div
          className="wds-global-navigation__dropdown-controls"
        >
          <Search
            inSearchModal={false}
            isSearchExpanded={false}
            model={
              Object {
                "search": "mock",
              }
            }
            onRedirectToSearchResults={[Function]}
            onSearchActivation={[Function]}
            onSearchClose={[Function]}
            onSearchSuggestionChosen={[Function]}
            onSearchSuggestionsImpression={[Function]}
            track={[Function]}
          />
          <User
            data={
              Object {
                "create-wiki": Object {
                  "create-wiki": "mock",
                },
                "logo": Object {
                  "href": "//fandom.wikia.com/",
                  "image-data": Object {
                    "name": "wds-company-logo-fandom-white",
                    "type": "wds-svg",
                  },
                  "tracking-label": "logo",
                  "type": "link-image",
                },
                "main-navigation": Array [
                  Object {
                    "href": "//fandom.wikia.com/topics/games",
                    "title": Object {
                      "key": "global-navigation-fandom-overview-link-vertical-games",
                      "type": "translatable-text",
                    },
                    "tracking-label": "link.games",
                    "type": "link-text",
                  },
                  Object {
                    "href": "//fandom.wikia.com/topics/games",
                    "title": Object {
                      "key": "global-navigation-fandom-overview-link-vertical-games",
                      "type": "translatable-text",
                    },
                    "tracking-label": "link.blah",
                    "type": "link-other",
                  },
                  Object {
                    "items": Array [
                      Object {
                        "href": "//fandom.wikia.com/explore",
                        "title": Object {
                          "key": "global-navigation-wikis-explore",
                          "type": "translatable-text",
                        },
                        "tracking-label": "link.explore",
                        "type": "link-text",
                      },
                    ],
                    "title": Object {
                      "key": "global-navigation-wikis-header",
                      "type": "translatable-text",
                    },
                    "tracking-label": "link.wikis",
                    "type": "link-group",
                  },
                ],
                "notifications": Object {
                  "notifications": "mock",
                },
                "partner-slot": null,
                "search": Object {
                  "search": "mock",
                },
                "services-domain": "https://services.wikia.com/",
                "user": null,
              }
            }
          />
          <div
            className="wds-global-navigation__start-a-wiki"
          >
            <LinkButton
              link={
                Object {
                  "create-wiki": "mock",
                }
              }
            />
          </div>
        </div>
        <div
          className="wds-global-navigation__modal-controls"
        >
          <SearchModal
            isOpen={false}
            openModal={[Function]}
          >
            <Search
              inSearchModal={true}
              isSearchExpanded={false}
              model={
                Object {
                  "search": "mock",
                }
              }
              onRedirectToSearchResults={[Function]}
              onSearchActivation={[Function]}
              onSearchClose={[Function]}
              onSearchSuggestionChosen={[Function]}
              onSearchSuggestionsImpression={[Function]}
              track={[Function]}
            />
            <nav
              className="wds-global-navigation__links"
            >
              <LinkText
                isStandaloneLink={true}
                key="0"
                link={
                  Object {
                    "href": "//fandom.wikia.com/topics/games",
                    "title": Object {
                      "key": "global-navigation-fandom-overview-link-vertical-games",
                      "type": "translatable-text",
                    },
                    "tracking-label": "link.games",
                    "type": "link-text",
                  }
                }
              />
              <LinkGroup
                key="2"
                link={
                  Object {
                    "items": Array [
                      Object {
                        "href": "//fandom.wikia.com/explore",
                        "title": Object {
                          "key": "global-navigation-wikis-explore",
                          "type": "translatable-text",
                        },
                        "tracking-label": "link.explore",
                        "type": "link-text",
                      },
                    ],
                    "title": Object {
                      "key": "global-navigation-wikis-header",
                      "type": "translatable-text",
                    },
                    "tracking-label": "link.wikis",
                    "type": "link-group",
                  }
                }
              />
            </nav>
          </SearchModal>
          <MobileUser
            data={
              Object {
                "create-wiki": Object {
                  "create-wiki": "mock",
                },
                "logo": Object {
                  "href": "//fandom.wikia.com/",
                  "image-data": Object {
                    "name": "wds-company-logo-fandom-white",
                    "type": "wds-svg",
                  },
                  "tracking-label": "logo",
                  "type": "link-image",
                },
                "main-navigation": Array [
                  Object {
                    "href": "//fandom.wikia.com/topics/games",
                    "title": Object {
                      "key": "global-navigation-fandom-overview-link-vertical-games",
                      "type": "translatable-text",
                    },
                    "tracking-label": "link.games",
                    "type": "link-text",
                  },
                  Object {
                    "href": "//fandom.wikia.com/topics/games",
                    "title": Object {
                      "key": "global-navigation-fandom-overview-link-vertical-games",
                      "type": "translatable-text",
                    },
                    "tracking-label": "link.blah",
                    "type": "link-other",
                  },
                  Object {
                    "items": Array [
                      Object {
                        "href": "//fandom.wikia.com/explore",
                        "title": Object {
                          "key": "global-navigation-wikis-explore",
                          "type": "translatable-text",
                        },
                        "tracking-label": "link.explore",
                        "type": "link-text",
                      },
                    ],
                    "title": Object {
                      "key": "global-navigation-wikis-header",
                      "type": "translatable-text",
                    },
                    "tracking-label": "link.wikis",
                    "type": "link-group",
                  },
                ],
                "notifications": Object {
                  "notifications": "mock",
                },
                "partner-slot": null,
                "search": Object {
                  "search": "mock",
                },
                "services-domain": "https://services.wikia.com/",
                "user": null,
              }
            }
            isOpen={false}
            openModal={[Function]}
            track={[Function]}
          />
          <Button
            className="wds-global-navigation__modal-control wds-global-navigation__modal-control-close"
            disabled={false}
            fullwidth={false}
            href={null}
            light={false}
            onClick={[Function]}
            secondary={false}
            square={false}
            text={true}
          >
            <Icon
              className=""
              name="close"
              small={false}
              tiny={false}
            />
          </Button>
        </div>
      </div>
    </div>
  </NotificationsDataProvider>
</ContextProvider>
`;
